src/domain/request/requestExecutor/RequestExecutorFactory.ts

```diff
@@ -0,0 +1,30 @@
+import { LoggerPort } from '@core/domain/common/port';
+import RequestExecutor from '@core/domain/common/port/RequestExecutorPort';
+import FetchDataUseCase from '@core/domain/common/usecase/FetchDataUseCase';
+import SelectDataUseCase from '@core/domain/common/usecase/SelectDataUseCase';
+import InvalidContentTypeError from '@core/domain/common/utils/error/InvalidContentTypeError';
+import UrlRequestExecutor from './UrlRequestExecutor';
+
+class RequestExecutorFactory  {
+    private requestExecutors: RequestExecutor[];
+
+    constructor(
+        fetchDataUseCase: FetchDataUseCase,
+        selectDataUseCase: SelectDataUseCase,
+        logger: LoggerPort,
+    ) {
+        this.requestExecutors = [new UrlRequestExecutor(fetchDataUseCase, selectDataUseCase, logger)];
+    }
+
+    create(contentType: string): RequestExecutor {
```

Comment:
Just cosmetic - it's not really a factory but a finder of existing stateless objects. The name just seems a bit misleading when you look at the implementation.

Reply:
Fair enough ,renaming from `Factory` to `StrategyFinder`